src/fsharp/CompileOps.fs

```diff
@@ -3752,28 +3752,29 @@ type TcAssemblyResolutions(tcConfig: TcConfig, results: AssemblyResolution list,
 //--------------------------------------------------------------------------
 
 let IsSignatureDataResource         (r: ILResource) = 
-    r.Name.StartsWith FSharpSignatureDataResourceName ||
-    r.Name.StartsWith FSharpSignatureDataResourceName2
+    r.Name.StartsWith(FSharpSignatureDataResourceName, StringComparison.Ordinal) ||
+    r.Name.StartsWith(FSharpSignatureDataResourceName2, StringComparison.Ordinal)
 
 let IsOptimizationDataResource      (r: ILResource) = 
-    r.Name.StartsWith FSharpOptimizationDataResourceName || 
-    r.Name.StartsWith FSharpOptimizationDataResourceName2
+    r.Name.StartsWith(FSharpOptimizationDataResourceName, StringComparison.Ordinal)|| 
+    r.Name.StartsWith(FSharpOptimizationDataResourceName2, StringComparison.Ordinal)
 
 let GetSignatureDataResourceName    (r: ILResource) = 
-    if r.Name.StartsWith FSharpSignatureDataResourceName then 
+    if r.Name.StartsWith(FSharpSignatureDataResourceName, StringComparison.Ordinal) then 
         String.dropPrefix r.Name FSharpSignatureDataResourceName
-    elif r.Name.StartsWith FSharpSignatureDataResourceName2 then 
+    elif r.Name.StartsWith(FSharpSignatureDataResourceName2, StringComparison.Ordinal) then 
         String.dropPrefix r.Name FSharpSignatureDataResourceName2
     else failwith "GetSignatureDataResourceName"
 
 let GetOptimizationDataResourceName (r: ILResource) = 
-    if r.Name.StartsWith FSharpOptimizationDataResourceName then 
+    if r.Name.StartsWith(FSharpOptimizationDataResourceName, StringComparison.Ordinal) then 
         String.dropPrefix r.Name FSharpOptimizationDataResourceName
-    elif r.Name.StartsWith FSharpOptimizationDataResourceName2 then 
+    elif r.Name.StartsWith(FSharpOptimizationDataResourceName2, StringComparison.Ordinal) then 
```

Comment:
I don't really know what comparison resource and assembly names should use. My understanding is that it doesn't actually make a difference if the thing being compared with is ASCII